addons/binding/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandler.java

```diff
@@ -0,0 +1,179 @@
+                vows_send.request(channelUID, Integer.toString(quantity.intValue()));
+            } else if (channelUID.getId().equals(ValloxMVBindingConstants.CHANNEL_HOME_AIR_TEMP_TARGET)) {
+                // Convert temperatur to milli degree Kelvin
+                QuantityType<Temperature> quantity = commandToQuantityType(command, SIUnits.CELSIUS)
+                        .toUnit(SmartHomeUnits.KELVIN);
+                int milliKelvin = quantity.multiply(new BigDecimal(100)).intValue();
+                vows_send.request(channelUID, Integer.toString(milliKelvin));
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    protected <U extends Quantity<U>> QuantityType<U> commandToQuantityType(Command command, Unit<U> defaultUnit) {
+        if (command instanceof QuantityType) {
+            return (QuantityType<U>) command;
+        }
+        return new QuantityType<U>(new BigDecimal(command.toString()), defaultUnit);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
+
+        String ip = getConfigAs(ValloxMVConfig.class).getIp();
+        vows_send = new ValloxMVWebSocket(ValloxMVHandler.this, ip);
+
+        BigDecimal bdUpdateInterval = getConfigAs(ValloxMVConfig.class).getUpdateinterval();
+        if (bdUpdateInterval == null || bdUpdateInterval.compareTo(new BigDecimal(15)) == -1) {
+            updateInterval = 60;
+        } else {
+            updateInterval = bdUpdateInterval.intValue();
+        }
+
+        try {
+            scheduleUpdates();
+        } catch (Exception e) {
+            logger.error("Error: {}", e.getMessage().toString());
+        }
+    }
+
+    private void scheduleUpdates() throws Exception {
+        logger.debug("Schedule vallox update every {} sec", updateInterval);
+
+        String ip = getConfigAs(ValloxMVConfig.class).getIp();
+        logger.debug("Connecting to ip: {}", ip);
+        // Open WebSocket
+        ValloxMVWebSocket vows = new ValloxMVWebSocket(ValloxMVHandler.this, ip);
+
+        updateTasks = scheduler.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                // Do a pure read request to websocket interface
+                vows.request(null, null);
+            }
+        }, 0, updateInterval, TimeUnit.SECONDS);
+    }
+
+    public void dataUpdated() {
+        lastUpdate = System.currentTimeMillis();
+    }
+
+    @Override
+    public void dispose() {
+        updateTasks.cancel(true);
+    }
+
+    @Override
+    protected void updateState(ChannelUID uid, State dt) {
+        super.updateState(uid, dt);
+    }
+
+    @Override
+    protected void updateStatus(ThingStatus ts, ThingStatusDetail statusDetail) {
+        super.updateStatus(ts, statusDetail);
+    }
+
+    @Override
+    protected void updateStatus(ThingStatus ts) {
+        super.updateStatus(ts);
+
+    }
+}
```

Comment:
Please add newlines to the end of all files, some (git) tools have problems if files do not contains a newline after the last data.

Reply:
Changed